lang/funcs/sensitive.go

```diff
@@ -0,0 +1,65 @@
+package funcs
+
+import (
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+)
+
+// SensitiveFunc returns a value identical to its argument except that
+// Terraform will consider it to be sensitive.
+var SensitiveFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name:             "value",
+			Type:             cty.DynamicPseudoType,
+			AllowUnknown:     true,
+			AllowNull:        true,
+			AllowMarked:      true,
+			AllowDynamicType: true,
+		},
+	},
+	Type: func(args []cty.Value) (cty.Type, error) {
+		// This function only affects the value's marks, so the result
+		// type is always the same as the argument type.
+		return args[0].Type(), nil
+	},
+	Impl: func(args []cty.Value, retType cty.Type) (ret cty.Value, err error) {
+		return args[0].Mark("sensitive"), nil
```

Comment:
```suggestion
                if !args[0].HasMark("sensitive") {
		    return args[0].Mark("sensitive"), nil
		}
```

Reply:
Ahh, reading these comments in order I saw this only after I wrote my first comment above :roll_eyes: ... I had a different change in mind here, like this:

```go
val, _ := args[0].Unmark()
return val.Mark("sensitive"), nil
```

The effect is the same with today's use of marks, but I'd prefer to make this function consistent in always using the same mark because I think this'll lead to fewer surprises down the road as we do maintenance. If there's a significant difference here I'm not seeing that ought to favor returning existing marks if present then please let me know!

Reply:
I pushed a new version with the logic I showed in my previous comment, because I feel better about this function _always_ using the mark `"sensitive"` even if some other part of Terraform is incorrectly using a different mark value, just so that the behavior of this function is relatively self-contained and won't vary based on oddities elsewhere.

I'm still happy to change it to what you proposed here if you have an argument for that particular approach that I'm not considering, though.